daemon/src/lib.rs

```diff
@@ -458,11 +465,25 @@ where
             .await?
     }
 
-    pub async fn reinitialise_wallet(&self, seed_words: &str) -> Result<()> {
+    pub async fn restore_wallet(&self, mnemonic: Mnemonic) -> Result<()> {
+        Seed::try_from(mnemonic.clone())?
+            .write_to(&self.wallet_seed_path)
+            .await?;
+
         self.wallet_actor_addr
-            .send(wallet::Reinitialise {
-                seed_words: seed_words.to_string(),
-            })
+            .send(wallet::Restore { mnemonic })
             .await?
     }
+
+    pub async fn backup_wallet(&self) -> Result<Mnemonic> {
+        let wallet_seed = Seed::read_from(&self.wallet_seed_path).await?;
+        let mnemonic = wallet_seed.try_into()?;
+        Ok(mnemonic)
+    }
+
+    pub fn generate_mnenomic(&self) -> Result<Mnemonic> {
+        let mut entropy = [0u8; 256];
+        rand::thread_rng().fill(&mut entropy);
+        Ok(Mnemonic::from_entropy(&entropy)?)
+    }
```

Comment:
This doesn't really use `self` so I don't think it needs to be on here? Also, I think `https://docs.rs/bip39/1.0.1/bip39/struct.Mnemonic.html#method.generate_in_with` might be a more appropriate function to use and we can just call this wherever needed and don't need to encapsulate it?

Reply:
I opted to do it the same as our seed generation because I did not know which random number generator i should use when calling `generare_in_with`

Reply:
Same as the one you are using currently, `thread_rng` is good enough :)